synapse/storage/databases/main/registration.py

```diff
@@ -2312,6 +2312,37 @@ async def add_refresh_token_to_user(
 
         return next_id
 
+    def _move_device_refresh_token_txn(
+        self, txn: LoggingTransaction, old_device_id: str, device_id: str
+    ) -> None:
+        """Updates rows of old_device_id with current device_id"""
+
+        self.db_pool.simple_update_txn(
+            txn,
+            "refresh_tokens",
+            {"device_id": old_device_id},
+            {"device_id": device_id},
+        )
```

Comment:
This is not safe -- device IDs are not globally unique and need to be scoped by user ID.